Add support for boolean fields in CsvImportMigrationBase::addFieldMapping()

mstenta opened the pull request:
This adds support for boolean field types in `CsvImportMigrationBase::addFieldMapping()`, which is the method used to dynmically add base and bundle fields to the default farmOS CSV importers. This means that bundle fields of type `boolean` will be automatically added, and base fields can be added by implementing `hook_farm_import_csv_base_fields()` (added in #916).

This PR builds on top of #916, so it includes all of that PR's commits as well. This PR is technically only one commit.

paul121 reviewed (commented):
+1 to supporting boolean fields... just trying to think how this works.

What values does it accept? are `true/false`, `null`, `0/1` all valid values? I hope an empty string `''` or `' '` doesn't result in populate the field as `false` ?

mstenta commented:
> +1 to supporting boolean fields... just trying to think how this works.
> 
> What values does it accept? are true/false, null, 0/1 all valid values?

We added this `boolean` plugin in #815. It uses PHP's built-in `filter_var($value, FILTER_VALIDATE_BOOLEAN)`.

This comment has some more info: https://github.com/farmOS/farmOS/pull/815#discussion_r1766870455

> I hope an empty string '' or ' ' doesn't result in populate the field as false ?

I think it results in the default value being used for that field. We use that to our advantage in the `is_movement` field of log importers (also added by PR #815), so `seeding` logs default to `TRUE` and `activity` logs default to `FALSE`, but you can optionally override that in the CSV.

I'll give it a quick test to confirm this...

mstenta commented:
> I'll give it a quick test to confirm this...

Tested leaving `is_group_assignment` blank on an `activity` log import, and it ends up as `FALSE` on the log.

mstenta commented:
Just remembered how we're handling this:

https://github.com/farmOS/farmOS/blob/31c50a55a949018f8726ffd4674e3286b04b229e/modules/core/migrate/src/Plugin/migrate/process/Boolean.php#L38-L42

So a boolean field that defaults to `TRUE` will be set to `TRUE` if the field is left blank in the CSV.
